streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -212,6 +215,28 @@
                                         CommonClientConfigs.METRICS_NUM_SAMPLES_DOC);
     }
 
+    // this is the list of configs for underlying clients
+    // that streams prefer different default values
+    private static final Map<String, Object> PRODUCER_DEFAULT_OVERRIDES;
+    static
+    {
+        Map<String, Object> tempProducerDefaultOverrides = new HashMap<>();
+        tempProducerDefaultOverrides.put(ProducerConfig.LINGER_MS_CONFIG, "100");
+
+        PRODUCER_DEFAULT_OVERRIDES = new HashMap<>(tempProducerDefaultOverrides);
```

Comment:
I was thinking that this should be `Collections.unmodifiableMap(tempProducerDefaultOverrides)` to ensure that it is never mutated after it's assigned.